src/smolvm/images/builder.py

```
@@ -57,17 +57,86 @@
+    for path in sorted((_GUEST_AGENT_CRATE_DIR / "src").glob("*.rs")):
+        hasher.update(path.relative_to(_REPO_ROOT).as_posix().encode())
+        hasher.update(path.read_bytes())
+    return hasher.hexdigest()
+
+
+def _guest_agent_binary() -> Path:
+    """Build and return the Rust guest-agent binary used by rootfs builders."""
+    target = _guest_agent_target_triple()
+    binary_path = _guest_agent_binary_path()
+    try:
+        subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)